### PR TITLE
BUG: GLM-Gaussian fit_constrained, wrong bse

### PR DESCRIPTION
**Pull request by josef-pkt**
issue #4193

This currently adds unit tests for Gaussian compared to OLS for a single parameter constraint, inlcudes cov_type='HC0' test.

There is one test failure with master, bug not yet fixed.
The work around is to assign cov_params_default which doesn't fail the unit tests.

cov_type='HC0' looks correct with `fit_constrained`, tests don't fail

bug fix which most likely works is
to assign directly to cov_params_default in fit_regularized

side issue: OLS and GLM don't agree on scale with master (in this branch), difference in example is around 1%
I got agreement between OLS and GLM-Gaussian for scale when I worked on top of #3856 which looks like fixes the scale difference.

Aside: fit_constrained never got a follow-up #1749, see comments in original PR #1714

### CODE REVIEW

**Comment by josef-pkt** [comment]
as red as expected

more tests: the Poisson and Logit/Binomial tests in discrete test_constrained don't include unit tests for cov_type

**Comment by josef-pkt** [comment]
correction: It looks like only discrete.Poisson has fit_constrained, it is not added to any other classes
unit tests are for GLM-Binomial not Logit, 
Stata results are for Logit and for GLM Binomial, one Stata result, `results_constraint2_robust`, has robust cov_type but is not yet used in unit tests

**Comment by josef-pkt** [comment]
question:
the original PR mentions convergence failures
last year I improved start_params in Poisson and better llnull handling.

Do the start_params changes affect or interact with fit_regularized? Is llnull correctly adjusted?

aside: Does llnull include an offset? 
It is not excluded explicitly in the creating the constrained model.
In that case, using offset in the unit tests will not produce the same llnull as the constrained model. 
How is the fixed part in the constraints that is reflected in transform offset accounted for in llnull computation?

**Comment by josef-pkt** [comment]
aside: GLM._fit_gradient does correctly a normalized_cov_params scale correction

```
        if rslt.normalized_cov_params is None:
            cov_p = None
        else:
            cov_p = rslt.normalized_cov_params / scale
```

**Comment by josef-pkt** [comment]
This should be green now, fixes scaling bug in GLM-Gaussian and makes the same change to the generic version and the Poisson method. 
(This is not DRYed out yet, same code in 3 places. One problem that remain is that we don't have a robust way to recreate Results instances based on an existing model.)

This should be finished for the bug fix and unit tests, but there are more minor refactorings and changes possible, for a later PR.

File: statsmodels/discrete/tests/test_constrained.py [hunk 1]
```diff
@@ -491,6 +491,34 @@ def test_fit_constrained_wrap(self):
...
+        cls.res2 = reslogit.results_constraint2_robust
+
+        mod1 = GLM(spector_data.endog, spector_data.exog,
+                   family=families.Binomial())
+
+        # not used to match Stata for HC
+        # nobs, k_params = mod1.exog.shape
+        # k_params -= 1   # one constraint
+        cov_type = 'HC0'
+        cov_kwds = {'scaling_factor': 32/31}
```
Comment (thequackdaddy) [inline statsmodels/discrete/tests/test_constrained.py]: You may need a `from __future__ import division` for this

**Comment by josef-pkt** [comment]
There ain't no future in python 2.7
amended and force pushed to add future division to discrete test module

**Comment by josef-pkt** [comment]
green enough, merging

**Pull request merged**